src/ImageSharp/Processing/Processors/Convolution/MedianRowOperation{TPixel}.cs

```diff
@@ -0,0 +1,164 @@
+        private readonly Rectangle bounds;
+        private readonly Buffer2D<TPixel> targetPixels;
+        private readonly Buffer2D<TPixel> sourcePixels;
+        private readonly KernelSamplingMap map;
+        private readonly int kernelSize;
+        private readonly bool preserveAlpha;
+
+        public MedianRowOperation(Rectangle bounds, Buffer2D<TPixel> targetPixels, Buffer2D<TPixel> sourcePixels, KernelSamplingMap map, int kernelSize, Configuration configuration, bool preserveAlpha)
+        {
+            this.bounds = bounds;
+            this.configuration = configuration;
+            this.targetPixels = targetPixels;
+            this.sourcePixels = sourcePixels;
+            this.map = map;
+            this.kernelSize = kernelSize;
+            this.preserveAlpha = preserveAlpha;
+            int kernelCount = this.kernelSize * this.kernelSize;
+            this.yChannelStart = kernelCount;
+            this.zChannelStart = this.yChannelStart + kernelCount;
+            this.wChannelStart = this.zChannelStart + kernelCount;
+        }
+
+        public void Invoke(int y, Span<Vector4> span)
+        {
+            // Span has kernelSize^2 followed by bound width.
+            int boundsLeft = this.bounds.Left;
+            int boundsWidth = this.bounds.Width;
+            int boundsRight = this.bounds.Right;
+            int kernelCount = this.kernelSize * this.kernelSize;
+            Span<Vector4> kernelBuffer = span.Slice(0, kernelCount);
+            Span<Vector4> channelVectorBuffer = span.Slice(kernelCount, kernelCount);
+            Span<Vector4> targetBuffer = span.Slice(kernelCount << 1, boundsWidth);
+
+            // Stack 4 channels of floats in the space of Vector4's.
+            Span<float> channelBuffer = MemoryMarshal.Cast<Vector4, float>(channelVectorBuffer);
+            var xChannel = channelBuffer.Slice(0, kernelCount);
+            var yChannel = channelBuffer.Slice(this.yChannelStart, kernelCount);
+            var zChannel = channelBuffer.Slice(this.zChannelStart, kernelCount);
+
+            var xOffsets = this.map.GetColumnOffsetSpan();
+            var yOffsets = this.map.GetRowOffsetSpan();
+            var baseXOffsetIndex = 0;
+            var baseYOffsetIndex = (y - this.bounds.Top) * this.kernelSize;
+
+            if (this.preserveAlpha)
+            {
+                for (var x = boundsLeft; x < boundsRight; x++)
+                {
+                    var index = 0;
+                    for (var w = 0; w < this.kernelSize; w++)
+                    {
+                        var j = yOffsets[baseYOffsetIndex + w];
+                        var row = this.sourcePixels.DangerousGetRowSpan(j);
+                        for (var z = 0; z < this.kernelSize; z++)
+                        {
+                            var k = xOffsets[baseXOffsetIndex + z];
+                            var pixel = row[k];
+                            kernelBuffer[index + z] = pixel.ToVector4();
```

Comment:
Perhaps we should Make `ReadOnlyKernel` `Kernel` and add setters. That way you can use a `DenseMatrix<T>` and use all the existing APIs for Kernel access.

Reply:
The `ConvolutionProcessor` uses a clever trick, where it builds up the target value row-by-row. This is allowed for `ReadOnlyKernel` type of `RowOperation` classes, where we're calculating weighted sums basically. These weighted sums can indeed be split arbitrary, as is done by clearing the target buffer before the loop and using `+=` in [line 152](https://github.com/SixLabors/ImageSharp/blob/af9bfe516eeded3e6a52a7e74cf531af86d00f56/src/ImageSharp/Processing/Processors/Convolution/ConvolutionProcessor%7BTPixel%7D.cs#L152)
However, the operation of finding a median, **cannot** be split up like this (or at least not without lots of caching). It mandates the traversal of the entire kernel in one pass, to get a single `Span` of pixel values to `Sort`. Hence the different logic here, compared to `ConvolutionProcessor`.

I'll update the code to use `Unsafe.Add` iso indexers.

Reply:
Ah no, I didn't mean the processors themselves I just meant the `ConvolutionState` struct. It's designed to track all the offsets you're manually tracking yourself. `ReadOnlyKernel` could become `Kernel` with a read/write indexer (maybe ref based).

Reply:
Made a separate MedianConvolutionState struct for this, as it's using `Vector4` iso `float`.